File: 8kyu/(8 kyu) Difference of Volumes of Cuboids/(8 kyu) Difference of Volumes of Cuboids.js
// 1 Plain Solution
function findDifference(a, b) {
    const volume1 = a[0] * a[1] * a[2];
    const volume2 = b[0] * b[1] * b[2];
    return Math.abs(volume1 - volume2);
}

// 2 Straightforward solution
function findDifference(a, b) {
    const volume = f => f[0] * f[1] * f[2];
    return Math.abs(volume(a) - volume(b));
}

// 3 Optimized solution
function findDifference(a, b) {
    const volume = f => f.reduce((a, c) => a * c);
    return Math.abs(volume(a) - volume(b));
}

// 4 Clever solution
const findDifference = (a, b) =>
    Math.abs(
        a.reduce((a, c) => a * c) -
        b.reduce((a, c) => a * c)
    );

// 5 Coding golf
const findDifference = (a, b) =>
    (
        f = (a,b) => a * b,
        Math.abs(a.reduce(f) - b.reduce(f))
    )

File: 8kyu/(8 kyu) Difference of Volumes of Cuboids/(8 kyu) Difference of Volumes of Cuboids.dart
// 1 Plain Solution
int findDifference(List<int> a, List<int> b) {
  final volume1 = a[0] * a[1] * a[2];
  final volume2 = b[0] * b[1] * b[2];
  return (volume1 - volume2).abs();
}

// 2 Straightforward solution
int findDifference(List<int> a, List<int> b) {
  final volume = (List<int> f) => f[0] * f[1] * f[2];
  return (volume(a) - volume(b)).abs();
}

// 3 Clever solution
int findDifference(List<int> a, List<int> b) {
  return (a.reduce((a, c) => a * c) - b.reduce((a, c) => a * c)).abs();
}

// 5 Coding golf
int findDifference(List<int> a, List<int> b) =>
    (a.reduce((a, c) => a * c) - b.reduce((a, c) => a * c)).abs();
